witness/src/Dscp/Resource/Keys/Functions.hs

```diff
@@ -64,6 +65,11 @@ storePath BaseKeyParams{..} appDir nodeNameP =
   where
     defPath = appDir </> (nodeNameP |+ ".key")
 
+-- | Generate key resources randomly.
+genRandomStore :: MonadIO m => m (KeyResources n)
+genRandomStore =
+    KeyResources . secretKeyDataFromPair <$> runSecureRandom keyGen
```

Comment:
Eeh, this is absolutely the same as `genStore Nothing`. I understand that you did this in order to reduce the constraints on this action, but then please either use this function in `genStore` or rewrite `genStore` so it accepts not a `Maybe`.

Reply:
With pleasure :tada: